src/groupByKey.ts

```diff
@@ -2,6 +2,19 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey(items: any[], key: string): any {
+  const groupedObj: GroupsMap<any> = {};
+
+  items.map((item) => {
```

Comment:
```suggestion
  items.forEach((item) => {
```
It is better to use forEach method, if you want only iterate through some array